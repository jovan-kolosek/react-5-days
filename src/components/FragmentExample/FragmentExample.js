import React, { Fragment } from 'react';
import RenderContext from '../ContextExample/ContextExample';

function FragmentExample() {
  // You can also use <> (empty tag) as a shortened syntax
  // Yet, this is more readable
  return (
    <RenderContext.Consumer>
    {
      toRender => {
        if (toRender === 'fragment') {
          return (
            <Fragment>
              <p>Some text.</p>
              <h2>A heading</h2>
              <p>More text.</p>
              <h2>Another heading</h2>
              <p>Even more text.</p>
            </Fragment>
          );
        } else {
          return null;
        }
      }
    }
    </RenderContext.Consumer>
   );
}

export default FragmentExample;
