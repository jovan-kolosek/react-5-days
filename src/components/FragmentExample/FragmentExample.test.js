import React from 'react';
import RenderContext from '../ContextExample/ContextExample';
import FragmentExample from './FragmentExample';
import renderer from 'react-test-renderer';

test('FragmentExample with default context renders', () => {
  const component = renderer.create(
    <FragmentExample />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});


test('FragmentExample wont render if context value not fragment' , () => {
  const component = renderer.create(
    <RenderContext.Provider value="portal">
      <FragmentExample />
    </RenderContext.Provider>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
