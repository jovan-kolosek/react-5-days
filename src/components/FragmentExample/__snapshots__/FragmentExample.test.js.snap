// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FragmentExample with default context renders 1`] = `
Array [
  <p>
    Some text.
</p>,
  <h2>
    A heading
</h2>,
  <p>
    More text.
</p>,
  <h2>
    Another heading
</h2>,
  <p>
    Even more text.
</p>,
]
`;

exports[`FragmentExample wont render if context value not fragment 1`] = `null`;
